test/new-e2e/tests/apm/vm_test.go

```diff
@@ -35,26 +49,29 @@ func vmSuiteOpts(tr transport, opts ...awshost.ProvisionerOption) []e2e.SuiteOpt
 
 // TestVMFakeintakeSuiteUDS runs basic Trace Agent tests over the UDS transport
 func TestVMFakeintakeSuiteUDS(t *testing.T) {
-	// FIXME: fix flakiness reaching the agent then enable the test
-	t.Skip("Skipping flaky test - incident-25117 AIT-9390")
-
 	cfg := `
 apm_config.enabled: true
 apm_config.receiver_socket: /var/run/datadog/apm.socket
 `
+	setupScript := `#!/bin/bash
+sudo mkdir -p /var/run/datadog
+sudo groupadd -r dd-agent
+sudo useradd -r -M -g dd-agent dd-agent
+sudo chown dd-agent:dd-agent /var/run/datadog`
+
+	options := vmSuiteOpts(uds,
+		// Create the /var/run/datadog directory and ensure
+		// permissions are correct so the agent can create
+		// unix sockets for the UDS transport
+		awshost.WithEC2InstanceOptions(ec2.WithUserData(setupScript)),
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Nice !